### DISCUSSION TITLE
Directory for save/load mixer channels setup is not memorized

### DISCUSSION THREAD

**Discussion by FredNor78** [jamulussoftware/jamulus#1742] [UI Related]
Jamulus 3.7.0 for Windows should rather memorize the directory used last time for the "Save/Load mixer channels setup" instead of constantly suggesting "C:\Windows\System32"

**Comment by hoffie** [comment]
> instead of constantly suggesting "C:\Windows\System32"

Hrm, I'm wondering where that comes from -- certainly nothing hardcoded within Jamulus source. Can other Windows users confirm that this is the default?

**Reply by FredNor78** [reply]
> certainly nothing hardcoded within Jamulus source

Agreed, but may be the last used directory could be stored in a Jamulus parameter file, in order to be suggested by default in the selection box next time we use the save/load function? The place is memorized during the session (by the embedded Microsoft Window mechanism) but lost when we leave Jamulus and restart it.

**Reply by dcorson-ticino-com** [reply]
For me on Win10 the suggested directory is the Desktop.
Another directory seems not to be stored.

**Reply by hoffie** [reply]
@dcorson-ticino-com Thanks for the input.

@FredNor78 I'm still wondering where it is coming from though. Do you use a custom Desktop shortcut? I think the current working directory can be changed there, could this be the source?

**Reply by FredNor78** [reply]
No particular custom Desktop shortcut for the client. 
The command line in the shortcut is "C:\Program Files\Jamulus\Jamulus.exe"

**Reply by hoffie** [reply]
I think there is a separate field for the working directory, but I'd assume it's empty or "C:\Program Files\Jamulus\" in your case?

**Reply by FredNor78** [reply]
Well, I discovered 2 different cases
1- If I start the Jamulus client from the Windows Applications drop list, the default directory for load/save is C:\Windows\System32
2- If I start the Jamulus client from my shortcut on the Desktop, the the default directory for load/save is the desktop

This in fact is coherent because if corresponds each time with the default directory on start

**Reply by FredNor78** [reply]
But, as a conclusion, I stay on my initial request, it would be interesting if we could, either specify a default jch files directory, or if at least, Jamulus would retain the last directory used, even after closing an reopening

**Reply by FredNor78** [reply]
I currently turnaround the situation on choosing the default directory on my desktop shortcut

![Capture d’écran 2021-05-22 213219](https://user-images.githubusercontent.com/81464717/119238851-7b421d00-bb45-11eb-95f9-b064bf2b480c.png)

Thanks for all. I am done
